tests/topotests/multicast-pim-sm-topo1/test_multicast_pim_sm_topo1.py

```diff
@@ -0,0 +1,1697 @@
+        "(*,G) and (S,G) got timeout from FRR3 node , verify using "
+        " 'show ip mroute' "
+    )
+
+    result = verify_ip_mroutes(
+        tgen, "f1", "*", IGMP_JOIN, "f1-r2-eth3", "f1-i8-eth2", expected=False
+    )
+    assert result is not True, "Testcase {} : Failed \n mroutes are"
+    " still present \n Error: {}".format(tc_name, result)
+    logger.info("Expected Behavior: {}".format(result))
+
+    step("IGMP groups are present verify using 'show ip igmp group'")
+    dut = "l1"
+    interface = "l1-i1-eth1"
+    result = verify_igmp_groups(tgen, dut, interface, IGMP_JOIN)
+    assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+    write_test_footer(tc_name)
+
+
+def test_modify_igmp_query_timer_p0(request):
+    """
+    TC_24:
+    Verify modification of IGMP query timer should get update
+    accordingly
+    """
+
+    tgen = get_topogen()
+    tc_name = request.node.name
+    write_test_header(tc_name)
+
+    # Creating configuration from JSON
+    kill_iperf(tgen)
+    clear_ip_mroute(tgen)
+    reset_config_on_routers(tgen)
+    clear_ip_pim_interface_traffic(tgen, topo)
+
+    # Don"t run this test if we have any failure.
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    step("Enable IGMP on FRR1 interface and send IGMP join (225.1.1.1)")
+    result = config_to_send_igmp_join_and_traffic(
+        tgen, topo, tc_name, "i1", "i1-l1-eth0", GROUP_RANGE, join=True
+    )
+    assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    result = iperfSendIGMPJoin(tgen, "i1", IGMP_JOIN, join_interval=1)
+    assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    step("Configure RP on R2 (loopback interface) for the" " group range 225.0.0.0/8")
+
+    input_dict = {
+        "r2": {
+            "pim": {
+                "rp": [
+                    {
+                        "rp_addr": topo["routers"]["r2"]["links"]["lo"]["ipv4"].split(
+                            "/"
+                        )[0],
+                        "group_addr_range": GROUP_RANGE,
+                    }
+                ]
+            }
+        }
+    }
+
+    result = create_pim_config(tgen, topo, input_dict)
+    assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+    step("Send multicast traffic from FRR3 to 225.1.1.1 receiver")
+    result = config_to_send_igmp_join_and_traffic(
+        tgen, topo, tc_name, "i2", "i2-f1-eth0", GROUP_RANGE, traffic=True
+    )
+    assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    result = iperfSendTraffic(tgen, "i2", IGMP_JOIN, 32, 2500)
+    assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+    step(
+        "Verify 'show ip mroute' showing correct RPF and OIF"
+        " interface for (*,G) and (S,G) entries on all the nodes"
+    )
+
+    sleep(5)
```

Comment:
Please add a comment explaining why we need to wait for 5 seconds.

Reply:
Thank you. Particular sleep statement is removed.